src/app/components/dataset-details/dataset-details.component.ts

```diff
@@ -70,4 +72,12 @@ export class ItemDetailsComponent {
   isNotEmtpy() {
     return !!this.datasetDetails && Object.keys(this.datasetDetails).length > 0
   }
+
+  getDistributions(): any[] {
+    let distributions = this.datasetDetails["resources"];
+    if (!!distributions && Array.isArray(distributions.value)) {
```

Comment:
Why do you need to check if distributions.value is an array?

Reply:
@brunopacheco1 
I tried to copy on how [other values are fetched](https://github.com/GenomicDataInfrastructure/gdi-userportal-frontend/blob/3f0e7218286fafa2945f09aad76489ac826a48dd/src/app/components/dataset-details/dataset-details.component.ts#L53-L54) in this component and i think this is done to prevent tyescript to fail because DataSet record type is set as generic key value of string or array(https://github.com/GenomicDataInfrastructure/gdi-userportal-frontend/blob/3f0e7218286fafa2945f09aad76489ac826a48dd/src/app/interfaces/dataset-details.interface.ts#L13)

Since we know the schema, the type should not be generic here. It is a considerable effort so I will recommend to add a refactoring ticket to add proper types everywhere